### ISSUE TITLE
Assertion hit trying to load config when submodules don't exist on disk

### ISSUE THREAD

**Issue by joshaber** [libgit2/libgit2#3414]
Unfortunately the repository that this reproduces on cannot be shared.

This is with abbe17ef1ffbc264101808829a3317c1c805df96.

```
Assertion failed: (error != 0), function submodule_load_from_config, file /Users/joshaber/Documents/Development/GitHub/objective-git/External/libgit2/src/submodule.c, line 1658.
0 libsystem_kernel.dylib 0x00007fff8b0ed286 __pthread_kill + 10
1 libsystem_c.dylib 0x00007fff827db9b3 abort + 129
2 libsystem_c.dylib 0x00007fff827a3a99 __assert_rtn + 321
3 org.libgit2.ObjectiveGit 0x0000000103542b87 submodule_load_from_config + 759
4 org.libgit2.ObjectiveGit 0x0000000103495370 git_config_backend_foreach_match + 288
5 org.libgit2.ObjectiveGit 0x00000001035450e4 git_config_file_foreach + 52
6 org.libgit2.ObjectiveGit 0x000000010353fad8 all_submodules + 600
7 org.libgit2.ObjectiveGit 0x000000010353f5ac git_submodule_foreach + 140
```

The repository has 17 submodules.

Two of the submodules didn't actually exist on disk. When we try to enumerate the submodules, we hit the above assertion and crash.

If I remove those two submodule entries, we don't hit that assertion anymore.

**Comment by carlosmn** [comment]
This may have been fixed with #3370. If you have a chance to test with that PR merged, that'd be great. Otherwise, we'll have to wait for that to make its way master and the app.

**Comment by joshaber** [comment]
Happily it _does_ seem to be fixed in #3370! :metal: 

Do you wanna close this, or wait for #3370 to land?
